cli/azd/extensions/azure.foundry.ai.agents/internal/cmd/listen.go

```diff
@@ -28,8 +28,11 @@ func newListenCommand() *cobra.Command {
 			defer azdClient.Close()
 
 			provider := project.NewAgentServiceTargetProvider(azdClient)
+			projectParser := &project.FoundryParser{AzdClient: azdClient}
 			host := azdext.NewExtensionHost(azdClient).
-				WithServiceTarget("foundry.agent", provider)
+				WithServiceTarget("foundry.agent", provider).
+				WithProjectEventHandler("preprovision", projectParser.SetIdentity).
+				WithProjectEventHandler("postdeploy", projectParser.CoboPostDeploy)
```

Comment:
I'd consider adding these as a service level event so we can handle these incrementally during deployment so that a transient failure on some other service does not stop the overall postdeploy from being invoked.

Reply:
I am using the original hook which was a project level hook for postdeploy.  It should be easy to switch in the future if needed. multi-service is actually not part of the spec